tests/functional/profiles/test_profile_dir.py

```diff
@@ -0,0 +1,159 @@
+import os
+import pytest
+import yaml
+from contextlib import contextmanager
+from pathlib import Path
+
+import dbt.flags as flags
+from dbt.tests.util import run_dbt_and_capture_stdout, write_file, rm_file
+
+
+@pytest.fixture(scope="class")
+def profiles_yml(profiles_root, dbt_profile_data):
+    write_file(yaml.safe_dump(dbt_profile_data), profiles_root, "profiles.yml")
+    return dbt_profile_data
+
+
+@pytest.fixture(scope="class")
+def profiles_home_root():
```

Comment:
Have you tested what happens if there is no profiles.yml at this location? Or the ~/.dbt directory doesn't exist? Because we can't depend on there being such a file in a test environment.

Reply:
I did for `profiles.yml`! I didn't check how `dbt debug` behaves if `~/.dbt` doesn't exist, but that's something I can follow-up on.

The change I made to `run_dbt` to allow `expect_pass=None` enabled the tests to run `dbt debug` and capture the output, even when there are no `profiles.yml` files.

The beautify of these tests is that they merely confirm that it is looking in the expected directories for `profiles.yml` -- it is irrelevant to these tests if the `profiles.yml` actually exist within the user-specified directory or not. Other tests should verify the correct behavior when `profiles.yml` does/doesn't exist.